### PR TITLE
WIP: Feature paymentorders first draft

### PR DESCRIPTION
**Pull request by NullableInt**
An attempt to add support for the PaymentOrders api

### CODE REVIEW

**Comment by cobraz** [comment]
🎉

Thanks for an awesome contribution!

You should run `yarn generate-docs` to generate some documentation. You don't have to do it before you are ready with this PR :)

**Comment by NullableInt** [comment]
Sure, I still has some work to do before I open this a pull request ready for review. I opened it as a draft so you and others can see what I've done and are planning to do.

**Comment by cobraz** [comment]
Understood, it looks great! :)

**Comment by NullableInt** [comment]
This is way harder than I thougt it would be. Setting up classes and such is a bit confusing. Is there a guide or reference for how to do it correctly? 😂

**Comment by asbjornu** [comment]
[Learning TypeScript](https://www.typescriptlang.org/docs/) may be a good idea, @NullableInt. 😃

**Comment by NullableInt** [comment]
> 
> 
> [Learning TypeScript](https://www.typescriptlang.org/docs/) may be a good idea, @NullableInt. 😃

Time to read!

File: src/Services/PaymentOrder/PaymentOrder.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { SwedbankBase } from '../../SwedbankBase';
+import { MerchantConfig } from '../../MerchantConfig';
+import { PaymentOrderResponse } from './models/PaymentOrderResponse';
+import { PaymentOrderRequest } from './models/PaymentOrderRequest';
+
+export class PaymentOrder extends SwedbankBase {
+  paymentOrder: PaymentOrder;
```
Comment (cobraz) [inline src/Services/PaymentOrder/PaymentOrder.ts:7]: How is this used?

File: src/Services/PaymentOrder/__tests__/PaymentOrder.tests.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import * as nock from 'nock';
+import { PaymentOrder } from "../PaymentOrder";
+import { PaymentOrderResponse } from "../models/PaymentOrderResponse";
```
Comment (cobraz) [inline src/Services/PaymentOrder/__tests__/PaymentOrder.tests.ts:3]:
```suggestion
```

File: src/Services/PaymentOrder/__tests__/PaymentOrder.tests.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    paymentorder = new PaymentOrder(merchantConfig);
+  });
+
+
+  it('should properly map properties', async () => {
+    nock('https://api.payex.com/psp')
+      .persist()
+      .post('/test')
+      .reply(200, PaymentOrderResponseExample);
+    let response = await paymentorder.get("/test");
```
Comment (cobraz) [inline src/Services/PaymentOrder/__tests__/PaymentOrder.tests.ts:23]:
```suggestion
    const response = await paymentorder.get("/test");
```

File: src/Services/PaymentOrder/models/PaymentOrderResponse.d.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { PaymentOrder } from "../PaymentOrder";
```
Comment (cobraz) [inline src/Services/PaymentOrder/models/PaymentOrderResponse.d.ts:1]:
```suggestion
```

File: src/Services/PaymentOrder/models/PaymentOrderResponse.d.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import { Operation } from "../../../models/Operation";
+import { Payment } from "../../CardPayment/models/Payment";
+
+export class PaymentOrderResponse {
```
Comment (cobraz) [inline src/Services/PaymentOrder/models/PaymentOrderResponse.d.ts:5]:
```suggestion
export interface PaymentOrderResponse {
```

Is this an interface?

File: tutorials/README.md [hunk 6]
```diff
@@ -2,4 +2,4 @@
 
 Our intentions with this is to make it as easy as possible to get started with our SDK, and ultimatly Swedbank Pay.
 
-- [Quickstart with Card Redirect Flow](quickstart-card-redirect.md.md)
+- [Quickstart with Card Redirect Flow](quickstart-card-redirect.md)
```
Comment (cobraz) [inline tutorials/README.md:5]: Nice catch 🎉

**Review by cobraz — APPROVED** [review]

File: src/Services/PaymentOrder/PaymentOrder.ts (hunk 1 above)
Comment (NullableInt) [reply src/Services/PaymentOrder/PaymentOrder.ts:7]: You ask good questions.

**Comment by cobraz** [comment]
@NullableInt What do you think, maybe we could merge this so we can start to iterate on it? :)

**Comment by NullableInt** [comment]
@cobraz sure, go ahead! I'll let you press the button 😄

File: README.md [hunk 7]
```diff
@@ -1,11 +1,13 @@
-![Build & Deploy](https://github.com/bjerkio/swedbank-pay-js/workflows/Build%20&%20Deploy/badge.svg)
```
Comment (cobraz) [inline README.md:1]: What's up with the diff?

**Review by cobraz — APPROVED** [review]
It looks good. There is a lot of changes caused by the fact that you are adding so many entities. I have some ideas that I'd like to discuss with you about the interfaces/entities if you are available.

**Comment by NullableInt** [comment]
> 
> 
> It looks good. There is a lot of changes caused by the fact that you are adding so many entities. I have some ideas that I'd like to discuss with you about the interfaces/entities if you are available.

Just tell me if you want a meeting, or contact me on slack!

**Pull request closed without merging**